pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -1697,13 +1697,16 @@ public static ByteBuf addBrokerEntryMetadata(ByteBuf headerAndPayload,
         //   | BROKER_ENTRY_METADATA_MAGIC_NUMBER | BROKER_ENTRY_METADATA_SIZE |         BROKER_ENTRY_METADATA         |
         //   |         2 bytes                    |       4 bytes              |    BROKER_ENTRY_METADATA_SIZE bytes   |
 
-        BrokerEntryMetadata brokerEntryMetadata = BROKER_ENTRY_METADATA.get();
+        BrokerEntryMetadata brokerEntryMetadata = BROKER_ENTRY_METADATA.get().clearBrokerTimestamp();
         for (BrokerEntryMetadataInterceptor interceptor : brokerInterceptors) {
             interceptor.intercept(brokerEntryMetadata);
             if (numberOfMessages >= 0) {
                 interceptor.interceptWithNumberOfMessages(brokerEntryMetadata, numberOfMessages);
             }
         }
+        if (!brokerEntryMetadata.hasBrokerTimestamp()) {
+            brokerEntryMetadata.setBrokerTimestamp(System.currentTimeMillis());
+        }
```

Comment:
The broker timestamp will be set in line:1702 if the `AppendBrokerTimestampMetadataInterceptor` is enabled. Is this PR wants to set the broker timestamp without `AppendBrokerTimestampMetadataInterceptor` enabled?

Reply:
Unless we can ensure that the `AppendBrokerTimestampMetadataInterceptor` cannot be disabled, this bug may always occur

Reply:
Ok, I think we can either make the AppendBrokerTimestampMetadataInterceptor enabled by default or remove the AppendBrokerTimestampMetadataInterceptor to always apply the broker timestamp.

Without this change, the broker timestamp will be enabled without AppendBrokerTimestampMetadataInterceptor. Essentially, we delete the functionality of the AppendBrokerTimestampMetadataInterceptor.

Reply:
I opened a discuss in dev thread https://lists.apache.org/thread/0vtjt436xvlg0fr4ogchkjdvjcp56mvz